--- mm2.7.pius/src/Makefile	Wed Jan 26 18:32:33 1994
+++ mm2.7/src/Makefile	Fri Nov 19 00:45:36 1999
@@ -38,7 +38,7 @@
 
 STATICFLAG=FOOBAR${HOST_ARCH}
 
-CFLAGS = -g -I.  ${$(STATICFLAG)} 
+CFLAGS = $(RPM_OPT_FLAGS) -I. -DLINUX ${$(STATICFLAG)} 
 # The following is better if you want to make sure you run with SYSV defined
 # CFLAGS = -g -I.  ${$(STATICFLAG)}  -DSYSV
 # Also, for SGI Irix, compile in K&R mode
@@ -72,10 +72,10 @@
 #	install -s -c $(LOCALBINDIR) $$file
 #
 #INSTALL = cp
-INSTALL = install -c -s
+INSTALL = install -c
 
 # Root of installation tree
-INSTROOT = /usr/local
+INSTROOT = /usr
 #
 # This is where binaries should be copied
 LOCALBINDIR = ${INSTROOT}/bin
@@ -91,7 +91,7 @@
 FONTSUFFIX=pcf
 
 # This is where the mailcap file should be put
-MAILCAPDIR = ${INSTROOT}/etc
+MAILCAPDIR = /etc
 
 # This helps with the "for" constructs below on some platforms:
 SHELL=/bin/sh
@@ -138,17 +138,6 @@
 	    echo "installing man page $$target";\
 	    ${CP} $$file $$target; 		\
 	done)
-	@(cd man;				\
-	for file in *.4;			\
-	do 					\
-	    target=$(MAN4DIR)/`basename $$file .4`.$(MAN4EXT); \
-	    echo "installing man page $$target";\
-	    ${CP} $$file $$target; 		\
-	done)
-	@echo "Renaming old mailcap file to $(MAILCAPDIR)/mailcap.old
-	-mv $(MAILCAPDIR)/mailcap $(MAILCAPDIR)/mailcap.old
-	@echo "installing file $(MAILCAPDIR)/mailcap"
-	${CP} mailcap $(MAILCAPDIR)
 
 uninstall-all:
 	@(cd bin; 				\
@@ -172,8 +161,6 @@
 	    echo "rm-ing man page $$target";\
 	    ${RM} $$target; 		\
 	done)
-	@echo "rm-ing file $(MAILCAPDIR)/mailcap"
-	@${RM} $(MAILCAPDIR)/mailcap
 
 clean:
 	-${RM} -rf *.BAK
