--- mm2.7/src/src/metamail/Makefile.am	2006-03-18 00:12:41.000000000 +0200
+++ mm2.7.ncurses3/src/src/metamail/Makefile.am	2006-03-18 00:12:20.000000000 +0200
@@ -1,7 +1,5 @@
 MAINTAINERCLEANFILES    = Makefile.in
 
-CFLAGS			= -lncurses
-
 INCLUDES                = -I$(top_builddir) -I$(top_builddir)/metamail \
                         -I$(top_srcdir) -I$(top_srcdir)/metamail
 
@@ -15,7 +13,8 @@
 mimencode_LDADD		= $(top_builddir)/metamail/libmetamail.la
 
 mailto_SOURCES		= mailto.c codes.c shared.c
-mailto_LDADD		= $(top_builddir)/metamail/libmetamail.la
+mailto_LDADD		= $(top_builddir)/metamail/libmetamail.la -ltinfo
+mailto_CFLAGS		= -I/usr/include/ncurses
 
 splitmail_SOURCES	= splitmail.c shared.c
 splitmail_LDADD		= $(top_builddir)/metamail/libmetamail.la
--- mm2.7/src/src/richmail/Makefile.am	2006-03-18 00:12:41.000000000 +0200
+++ mm2.7.ncurses3/src/src/richmail/Makefile.am	2006-03-18 00:12:20.000000000 +0200
@@ -1,14 +1,13 @@
 MAINTAINERCLEANFILES    = Makefile.in
 
-CFLAGS			= -lncurses
-
 INCLUDES                = -I$(top_builddir) -I$(top_builddir)/metamail \
                         -I$(top_srcdir) -I$(top_srcdir)/metamail
 
 bin_PROGRAMS            = richtext richtoatk
 
 richtext_SOURCES	= richtext.c richlex.c richset.c usascii.c iso2022.c
-richtext_LDADD		= $(top_builddir)/metamail/libmetamail.la
+richtext_LDADD		= $(top_builddir)/metamail/libmetamail.la -ltinfo
+richtext_CFLAGS		= -I/usr/include/ncurses
 
 richtoatk_SOURCES	= richtoatk.c richlex.c richset.c usascii.c iso2022.c
 richtoatk_LDADD		= $(top_builddir)/metamail/libmetamail.la
